### PR TITLE
feat: Add read partition support

### LINKED ISSUE #223: Reading one partition data.
The table api should return one partition's data in arrow format.

### PR DESCRIPTION
**Pull request by liurenjie1024**
Close #223

### CODE REVIEW

**Comment by liurenjie1024** [comment]
CC @Xuanwo @ZENOTME

File: icelake/src/types/on_disk/partition_spec.rs [hunk 1]
```diff
@@ -6,6 +6,7 @@ use crate::Result;
 
 /// Parse schema from json bytes.
 pub fn parse_partition_spec(bs: &[u8]) -> Result<types::PartitionSpec> {
+    println!("parse_partition_spec: {:?}", String::from_utf8_lossy(bs));
```
Comment (ZENOTME) [inline icelake/src/types/on_disk/partition_spec.rs:9]: remove this?

**Review by ZENOTME — APPROVED** [review]
LGTM

**Review by Xuanwo — APPROVED** [review]
Cool!

**Pull request merged**